playbook/app/pb_kits/playbook/pb_multi_level_select/_multi_level_select.tsx

```diff
@@ -232,9 +249,35 @@ const MultiLevelSelect = (props: MultiLevelSelectProps) => {
     }
   }
 
+  // Single select
+  const handleRadioButtonClick = (
+    e: React.ChangeEvent<HTMLInputElement>,
+  ) => {
+    const { id: selectedItemID, value: inputText } = e.target
+    // Reset tree checked state, triggering useEffect
+    const treeWithNoSelections = modifyRecursive(formattedData, false)
+    // Update tree with single selection
+    const treeWithSelectedItem = modifyValue(selectedItemID, treeWithNoSelections, true)
+    const selectedItem = filterFormattedDataById(treeWithSelectedItem, selectedItemID)
+
+    setSingleSelectInputValue(inputText)
+    // Reset the filter to always display dropdown options on click
+    setFilterItem("")
+    setIsClosed(true)
+
+    onSelect(selectedItem)
+  };
+
+  // Single select: reset the tree state upon typing
+  const handleRadioInputChange = (inputText: string) => {
+    modifyRecursive(formattedData, false)
+    setSingleSelectInputValue(inputText)
+    setFilterItem(inputText)
+  };
+
```

Comment:
This is the heart of the logic.

Upon a radio selection via the dropdown:
- Uncheck everything in the tree via `modifyRecursive`
- Find the selected item via `modifyValue`, and check it. None of its children should be selected because only one option can ever be selected
- Display the value via the `setSingleSelectInputValue`
- Reset the filter - per the mock, all dropdown options should be shown
- Close the dropdown
- Call `onSelect`

Upon text input change:
- Reset the tree and uncheck everything
- TLDR: don't break type ahead

Reply:
looks good @gavhuang ! Thank you for all the comments explaining your thought processes!